encoding_converter.cpp

```diff
@@ -49,7 +50,7 @@ std::string EncodingConverter::convert(const std::string& str){
     size_t output_left = buffer_size;
     size_t input_left = str.size();
     size_t result = iconv(iconv_handler, &working_input, &input_left, &working_output, &output_left);
-    if(result == (size_t)-1){
+    if(result == std::numeric_limits<size_t>::max()){
```

Comment:
As the doc of iconv says `In case of error,  it sets errno and returns (size_t) -1.`, maybe it's better to use `size_t(-1)`